### PR TITLE
feat: add frequency in ui

### LINKED ISSUE #153: [Sub][UI] Create frequency component

### PR DESCRIPTION
**Pull request by janfuhrer**
closes #153 & #154

### CODE REVIEW

File: src/main/kotlin/ch/bfh/habits/services/HabitService.kt [hunk 1]
```diff
@@ -51,9 +51,6 @@ class HabitService(private val habitDAO: HabitDAO) {
 
     @Transactional
     fun updateHabitById(id: Long, habitDTO: HabitDTO) {
-        if (habitDTO.id != id) {
```
Comment (Andy-Bigler) [inline src/main/kotlin/ch/bfh/habits/services/HabitService.kt:54]: We do not need this?

File: ui/lib/interfaces.ts [hunk 2]
```diff
@@ -7,6 +7,8 @@ export interface BaseRecord {
 export interface Habit extends BaseRecord {
   title: string
   description: string
+  frequency: string
```
Comment (Andy-Bigler) [inline ui/lib/interfaces.ts:10]: Create a type for this in the frontend (not an enum):
`export type frequency = 'DAILY | 'WEEKLY' | 'MONTHLY' | 'YEARLY'`

File: src/main/kotlin/ch/bfh/habits/services/HabitService.kt (hunk 1 above)
Comment (janfuhrer) [reply src/main/kotlin/ch/bfh/habits/services/HabitService.kt:54]: updating a habit is not working at the moment.
The UI only sends the id in the path and not in the body of the request.
So we can discuss whether to send the ID inside the body or remove it in the backend.

**Comment by janlauber** [comment]
rebased

**Pull request merged**